<launch>

    <arg name="is_sim" default="false"/>

    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find robot1_controller)/config/trajectory_controllers.yaml" command="load"/>

    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find robot1_controller)/config/joint_state_controller.yaml" command="load"/>

    <param name="robot_description" textfile="$(find robot1_description)/urdf/robot1.urdf" />

    <!-- If interfacing with the real Hardware -->
    <group unless="$(arg is_sim)">
        <!-- Launch the angles converter service -->
        <node pkg="robot1_controller" type="angles_converter_cpp" name="angles_converter" 
            respawn="true" output="screen"/>

        <!-- Launch the Hardware interface with the real robot -->
        <node name="robot1_interface_node" pkg="robot1_controller" type="robot1_interface_node" 
            ns="/robot1" respawn="true" output="screen"/>
    </group>

    <!-- Load the controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" 
        ns="/robot1" args="joint_state_controller arm_controller gripper_controller" />

    <!-- Convert joint states to TF transforms for rviz, etc -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
        <remap from="/joint_states" to="/robot1/joint_states" />
    </node>

</launch>